packages/build-tools/src/steps/functions/sendSlackMessage.ts

```diff
@@ -0,0 +1,73 @@
+import { BuildFunction, BuildStepInput, BuildStepInputValueTypeName } from '@expo/steps';
+import fetch, { Response } from 'node-fetch';
+import { bunyan } from '@expo/logger';
+
+export function createSendSlackMessageFunction(): BuildFunction {
+  return new BuildFunction({
+    namespace: 'eas',
+    id: 'send_slack_message',
+    name: 'Send Slack message',
+    inputProviders: [
+      BuildStepInput.createProvider({
+        id: 'message',
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+        required: true,
+      }),
+      BuildStepInput.createProvider({
+        id: 'slack_hook_url',
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+        required: false,
+      }),
+    ],
+    fn: async (stepCtx, { inputs, env }) => {
+      const { logger } = stepCtx;
+      const slackMessage = inputs.message.value as string;
+      const slackHookUrl = (inputs.slack_hook_url.value as string) ?? env.SLACK_HOOK_URL;
+      await sendSlackMessageAsync({ logger, slackHookUrl, slackMessage });
+    },
+  });
+}
+
+async function sendSlackMessageAsync({
+  logger,
+  slackHookUrl,
+  slackMessage,
+}: {
+  logger: bunyan;
+  slackHookUrl: string | undefined;
+  slackMessage: string;
+}): Promise<void> {
+  if (!slackHookUrl) {
+    logger.warn(
+      'Slack webhook URL not provided - provide input "slack_hook_url" or set "SLACK_HOOK_URL" secret'
+    );
+    throw new Error(
+      'Sending Slack message failed - provide input "slack_hook_url" or set "SLACK_HOOK_URL" secret'
+    );
+  }
```

Comment:
nit: My vote would be to put the validation in the `BuildFunction`, so `sendSlackMessageAsync` is only concerned about business and it assumes valid input. Not that it matters here, just my opinion.